fabric-particles-v1/src/main/java/net/fabricmc/fabric/impl/particles/ParticleRegistryImpl.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.particles;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.particles.ParticleRegistry;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.particle.ParticleFactory;
+import net.minecraft.particle.DefaultParticleType;
+import net.minecraft.particle.ParticleEffect;
+import net.minecraft.particle.ParticleType;
+
+import java.util.HashMap;
+
+/** Core methods for registering particles with the Fabric API. */
+public class ParticleRegistryImpl implements ParticleRegistry {
+	public DefaultParticleType createSimpleParticleType() { return createSimpleParticleType(false); }
+	public DefaultParticleType createSimpleParticleType(boolean shouldAlwaysSpawn) {
+		return new DefaultParticleType(shouldAlwaysSpawn) {};
```

Comment:
A small comment to explain why you need an anonymous class wouldn't hurt IMO

Reply:
I can add a comment if it's really needed, but it's really just to bypass a protected constructor. Same for the other anonymous class in that file, nothing particularly interesting.

Reply:
It's about maintainability, it's nice to be able to browse the sources and not have to stay stuck for 5mn wondering why it's using a sneaky anonymous class.

Reply:
Fair enough, I'll add that.